internal/clientcache/cmd/daemon/addtoken.go

```diff
@@ -76,38 +77,48 @@ func (c *AddTokenCommand) Run(args []string) int {
 		return base.CommandCliError
 	}
 
-	apiErr, err := c.Add(ctx, client, keyringType, tokenName)
+	resp, apiErr, err := c.Add(ctx, client, keyringType, tokenName)
 	if err != nil {
 		c.PrintCliError(err)
 		return base.CommandCliError
 	}
 	if apiErr != nil {
 		c.PrintApiError(apiErr, "Error from daemon when adding a token")
 		return base.CommandApiError
-
+	}
+	switch base.Format(c.UI) {
+	case "json":
+		if ok := c.PrintJsonItem(resp); !ok {
+			return base.CommandCliError
+		}
+	case "table":
+		c.UI.Output("The daemon add-token operation completed successfully.")
 	}
 	return base.CommandSuccess
 }
 
-func (c *AddTokenCommand) Add(ctx context.Context, client *api.Client, keyringType, tokenName string) (*api.Error, error) {
-	const op = "daemon.(AddTokenCommand).Add"
-
+func (c *AddTokenCommand) Add(ctx context.Context, apiClient *api.Client, keyringType, tokenName string) (*api.Response, *api.Error, error) {
 	pa := daemon.UpsertTokenRequest{
-		BoundaryAddr: client.Addr(),
+		BoundaryAddr: apiClient.Addr(),
 	}
 	switch keyringType {
 	case "", base.NoneKeyring:
-		token := client.Token()
+		token := apiClient.Token()
 		if parts := strings.SplitN(token, "_", 4); len(parts) == 3 {
 			pa.AuthTokenId = strings.Join(parts[:2], "_")
 		} else {
-			return nil, errors.New(ctx, errors.InvalidParameter, op, "found auth token is not in the proper format")
+			return nil, nil, errors.New("The found auth token is not in the proper format.")
+		}
+		if c.FlagOutputCurlString {
+			// do not output the actual auth token if we are printing out the curl string
+			pa.AuthToken = "$BOUNDARY_TOKEN"
```

Comment:
Please look at how this is done in the normal client via `boundary config get-token`